tools/lint_diff.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import os
+import sys
+import subprocess
+
+CONFIG = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)),
+    'lint_diff.ini',
+)
+
+
+def rev_list(branch, num_commits):
+    """List commits in reverse chronological order.
+
+    Only the first `num_commits` are shown.
+
+    """
+    res = subprocess.run(
+        [
+            'git',
+            'rev-list',
+            '--max-count',
+            f'{num_commits}',
+            '--first-parent',
+            branch
+        ],
+        capture_output=True,
+        text=True,
+    )
+    res.check_returncode()
+    return res.stdout.rstrip('\n').split('\n')
+
+
+def find_branch_point():
+    """Find when the current branch split off master.
+
+    It is based off of this Stackoverflow post:
+
+    https://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git#4991675
+
+    """
+    branch_commits = rev_list('HEAD', 1000)
+    master_commits = set(rev_list('master', 1000))
```

Comment:
there is probably an incantation using `git merge-base`?

Reply:
I believe they are different based on https://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git#4991675.